amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ConnectionContainer.java

```diff
@@ -63,14 +61,14 @@ public void unLoad(NamespaceBundle namespaceBundle) {
                     connectionMap.remove(namespaceName);
                 }
 
-                if (ExchangeContainer.getExchangeMap().containsKey(namespaceName)) {
-                    ExchangeContainer.getExchangeMap().get(namespaceName).clear();
-                    ExchangeContainer.getExchangeMap().remove(namespaceName);
+                if (exchangeContainer.getExchangeMap().containsKey(namespaceName)) {
+                    exchangeContainer.getExchangeMap().get(namespaceName).clear();
+                    exchangeContainer.getExchangeMap().remove(namespaceName);
```

Comment:
here, why clear() firstly and then remove(namespaceName)?

Reply:
I seems to make the map can be  included in gc.

Reply:
If you want/need to call `clear()` on the `Set<AmqpConnection>` when it's removed from the map, you could do this in one line.

```java
exchangeContainer.getExchangeMap().remove(namespaceName).clear();
```

Remove returns the object it removed, so you can append the clear call.

I'm not sure that's needed.  The set won't be accessable anymore, so it will be garbage collected, and so will any connections that it holds.